File: src/Maximize Number of 1's/Code/solution.js
class Solution {
  maxOnes(arr, k) {
    let left = 0;
    let zeroCount = 0;
    let maxLen = 0;

    for (let right = 0; right < arr.length; right++) {
      // If we see a zero, increase count
      if (arr[right] === 0) {
        zeroCount++;
      }

      // Shrink window if zero count > k
      while (zeroCount > k) {
        if (arr[left] === 0) {
          zeroCount--;
        }
        left++;
      }

      // Update max length
      maxLen = Math.max(maxLen, right - left + 1);
    }

    return maxLen;
  }
}

File: src/Maximize Number of 1's/Code/solution.cpp
class Solution
{
public:
    int maxOnes(vector<int> &arr, int k)
    {
        int left = 0;
        int zeroCount = 0;
        int maxLen = 0;

        for (int right = 0; right < arr.size(); right++)
        {
            // If current element is 0, increase zero count
            if (arr[right] == 0)
            {
                zeroCount++;
            }

            // If zeros exceed k, shrink the window
            while (zeroCount > k)
            {
                if (arr[left] == 0)
                {
                    zeroCount--;
                }
                left++;
            }

            // Update maximum window length
            maxLen = max(maxLen, right - left + 1);
        }

        return maxLen;
    }
};
